docs/contributing/testing/naming-conventions.mdx

```diff
@@ -0,0 +1,254 @@
+# ​Naming conventions
+
+:::caution
+
+The information in this document is a work in progress. Decisions need to be discussed and feedback
+needs to be collected before any of the information in here can be considered official.
+
+:::
+
+The main goal of the name is to describe the test in a way that makes it easy to understand what the
+test does and which cases it is intending to cover. A good list of test names gives developers the
+ability to quickly review that the suite covers all important use cases, and to easily identify gaps
+when it does not.
+
+There are many ways of describing a test but what most methods have in common is that they include
+the following
+[[1]](https://dzone.com/articles/7-popular-unit-test-naming)[[2]](https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea)[[3]](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests):
+
+- Unit under test
+- State under test
+- Expected behavior
+
+We've therefore chosen to use the following format (`[]` means optional)
+
+`<unit> [given <prerequisite>] should <behavior> when <state>`
```

Comment:
Per my earlier comment about framework-specific examples, I often default to my C# experience and the Microsoft examples of roughly "given-when-then". Their example:

`Add_MultipleNumbers_ReturnsSumOfNumbers`

isn't precisely that but connects with @coroiu's (1) above.

Are Bitwarden engineers finding a lot of prerequisites defined for tests that wouldn't fit into "state"?

Reply:
To be honest I would've suggested `MethodName_StateUnderTest_ExpectedBehavior` if it wasn't so different from what we already have in this repo, it's my favorite convention from my C#-heavy days. 

I actually prefer `<unit> [given -] when - then` because I think it's so similar to `MethodName_StateUnderTest_ExpectedBehavior`. It's basically just the last two flippe `<unit = MethodName> should <ExpectedBehavior> when <StateUnderTest>`

Reply:
NOTE: As per previous discussion with @eliykat `should` is no longer part of the name.